app/src/main/java/dev/msfjarvis/lobsters/data/repo/LobstersRepository.kt

```diff
@@ -2,35 +2,56 @@ package dev.msfjarvis.lobsters.data.repo
 
 import dev.msfjarvis.lobsters.data.local.LobstersPost
 import dev.msfjarvis.lobsters.database.LobstersDatabase
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 class LobstersRepository @Inject constructor(private val lobstersDatabase: LobstersDatabase) {
 
+  private val savedPostsCache: MutableMap<String, LobstersPost> = mutableMapOf()
+  private val coroutineScope = CoroutineScope(Job() + Dispatchers.IO)
+
+  init {
+    coroutineScope.launch {
+      getAllPosts().forEach {
+        savedPostsCache.putIfAbsent(it.short_id, it)
+      }
+    }
+  }
```

Comment:
@msfjarvis currently i am creating cached map on class init. However we can also do it in the first DB access. What are your thoughts on it?

Reply:
Caching on init is fine, we want it to be immediately available